distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -1194,7 +1193,15 @@ type Table
              example_to_xlsx = Examples.inventory_table.write (enso_project.data / "example_xlsx_output.xlsx") Excel
     write : File|Text -> File_Format -> Existing_File_Behavior -> Match_Columns -> Problem_Behavior -> Nothing ! Column_Mismatch | Illegal_Argument_Error | File_Not_Found | IO_Error
     write self path format=Auto_Detect on_existing_file=Existing_File_Behavior.Backup match_columns=Match_Columns.By_Name on_problems=Report_Warning =
-        format.write_table (File.new path) self on_existing_file match_columns on_problems
+        file = File.new path
+        case format of
+            _ : Auto_Detect ->
+                base_format = format.get_format file
+                if base_format == Nothing then Error.throw (Errors.Unsupported_File_Type_Data ("No File_Format supports '" + file.extension + "'")) else
+                    Panic.catch Errors.No_Such_Method_Error_Data (base_format.write_table file self on_existing_file match_columns on_problems) _->
+                       name = Meta.get_constructor_name (Meta.meta base_format).constructor
+                       Error.throw (Errors.Illegal_Argument_Error_Data ("Saving a Table as " + name + " is not supported."))
+            _ -> format.write_table file self on_existing_file match_columns on_problems
```

Comment:
Shouldn't this branch handle the `No_Such_Method_Error_Data` too? The user can still provide some custom format that is unsupported and if we can get them a nice error message for Auto, let's also give it for this case.